plugins/deisa/deisa.cxx

```diff
@@ -0,0 +1,206 @@
+/******************************************************************************
+ * Copyright (c) 2020-2024 Centre national de la recherche scientifique (CNRS)
+ * Copyright (c) 2020-2024 Commissariat a l'énergie atomique et aux énergies alternatives (CEA)
+ * Copyright (c) 2020-2023 Institut national de recherche en informatique et en automatique (Inria)
+ * Copyright (c) 2020-2024 Université Paris-Saclay
+ * Copyright (c) 2020-2024 Université de Versailles Saint-Quentin-en-Yvelines
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ *****************************************************************************/
+
+#include <memory>
+#include <memory_resource>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <pybind11/embed.h>
+#include <pybind11/numpy.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include <mpi.h>
+#include <pdi.h>
+#include <pdi/array_datatype.h>
+#include <pdi/context.h>
+#include <pdi/data_descriptor.h>
+#include <pdi/datatype.h>
+#include <pdi/expression.h>
+#include <pdi/logger.h>
+#include <pdi/paraconf_wrapper.h>
+#include <pdi/plugin.h>
+#include <pdi/python/tools.h>
+#include <pdi/ref_any.h>
+#include <pdi/scalar_datatype.h>
+
+namespace {
+
+namespace py = pybind11;
+
+using namespace PDI;
+using namespace py::literals;
+
+/** The deisa plugin
+ */
+class deisa_plugin: public Plugin
+{
+	static constexpr char DEISA_COMPATIBLE_VERSION[] = "0.3.2"; // Used to check compatibility with Deisa's python library
+
+	Expression m_scheduler_info;
+	std::unordered_map<std::string, Datatype_template_sptr> m_deisa_arrays;
```

Comment:
Could you make these two into temporary variables that you pass as a parameter to `init_deisa`

Reply:
My bad, `init_deisa` is called in a callback